lib/babe/crypto.go

```diff
@@ -83,6 +84,40 @@ func checkPrimaryThreshold(randomness Randomness,
 	return inoutUint.Compare(threshold) < 0, nil
 }
 
+func claimSecondarySlot(randomness Randomness,
+	slot, epoch uint64,
+	authorities []types.Authority,
+	threshold *scale.Uint128,
+	keypair *sr25519.Keypair,
+) (*VrfOutputAndProof, error) {
+
+	secondarySlotAuthor, err := getSecondarySlotAuthor(slot, len(authorities), randomness)
+	if err != nil {
+		return nil, err
+	}
+
+	for authority_index := range authorities {
+		if authority_index == int(secondarySlotAuthor) {
+			transcript := makeTranscript(randomness, slot, epoch)
+
+			out, proof, err := keypair.VrfSign(transcript)
+			if err != nil {
+				return nil, err
+			}
+
+			logger.Debugf("claimed secondary slot, for slot number: %d", slot)
+
+			return &VrfOutputAndProof{
+				output: out,
+				proof:  proof,
+			}, nil
+		}
+	}
```

Comment:
this is where your error is coming from - this doesn't actually check if this node is allowed to produce a secondary block. `authorities` is the list of all the current authorities, this is returning an authorship proof if anyone is allowed to produce a secondary block.

you need to pass in our `authorityIndex` into this function and check if `secondarySlotAuthor == authorityIndex`.